drivers/video/ov7725.c

```diff
@@ -0,0 +1,651 @@
+};
+
+static struct ov7725_reg ov7725_init_reg_tb[] = {
+	/*Output config*/
+	{ OV7725_CLKRC,          0x00 },
+	{ OV7725_COM7,           0x06 },
+	{ OV7725_HSTART,         0x3f },
+	{ OV7725_HSIZE,          0x50 },
+	{ OV7725_VSTART,         0x03 },
+	{ OV7725_VSIZE,          0x78 },
+	{ OV7725_HREF,           0x00 },
+	{ OV7725_HOUTSIZE,       0x50 },
+	{ OV7725_VOUTSIZE,       0x78 },
+
+	/*DSP control*/
+	{ OV7725_TGT_B,          0x7f },
+	{ OV7725_FIXGAIN,        0x09 },
+	{ OV7725_AWB_CTRL0,      0xe0 },
+	{ OV7725_DSP_CTRL1,      0xff },
+	{ OV7725_DSP_CTRL2,      0x00 },
+	{ OV7725_DSP_CTRL3,      0x00 },
+	{ OV7725_DSP_CTRL4,      0x00 },
+
+	/*AGC AEC AWB*/
+	{ OV7725_COM8,           0xf0 },
+	{ OV7725_COM4,           0x81 },
+	{ OV7725_COM6,           0xc5 },
+	{ OV7725_COM9,           0x11 },
+	{ OV7725_BDBASE,         0x7F },
+	{ OV7725_BDMSTEP,        0x03 },
+	{ OV7725_AEW,            0x40 },
+	{ OV7725_AEB,            0x30 },
+	{ OV7725_VPT,            0xa1 },
+	{ OV7725_EXHCL,          0x9e },
+	{ OV7725_AWB_CTRL3,      0xaa },
+	{ OV7725_COM8,           0xff },
+
+	/*matrix shapness brightness contrast*/
+	{ OV7725_EDGE1,          0x08 },
+	{ OV7725_DNSOFF,         0x01 },
+	{ OV7725_EDGE2,          0x03 },
+	{ OV7725_EDGE3,          0x00 },
+	{ OV7725_MTX1,           0xb0 },
+	{ OV7725_MTX2,           0x9d },
+	{ OV7725_MTX3,           0x13 },
+	{ OV7725_MTX4,           0x16 },
+	{ OV7725_MTX5,           0x7b },
+	{ OV7725_MTX6,           0x91 },
+	{ OV7725_MTX_CTRL,       0x1e },
+	{ OV7725_BRIGHT,         0x08 },
+	{ OV7725_CNST,           0x20 },
+	{ OV7725_UVADJ0,         0x81 },
+	{ OV7725_SDE,            0X06 },
+	{ OV7725_USAT,           0x65 },
+	{ OV7725_VSAT,           0x65 },
+	{ OV7725_HUECOS,         0X80 },
+	{ OV7725_HUESIN,         0X80 },
+
+	/*GAMMA config*/
+	{ OV7725_GAM1,           0x0c },
+	{ OV7725_GAM2,           0x16 },
+	{ OV7725_GAM3,           0x2a },
+	{ OV7725_GAM4,           0x4e },
+	{ OV7725_GAM5,           0x61 },
+	{ OV7725_GAM6,           0x6f },
+	{ OV7725_GAM7,           0x7b },
+	{ OV7725_GAM8,           0x86 },
+	{ OV7725_GAM9,           0x8e },
+	{ OV7725_GAM10,          0x97 },
+	{ OV7725_GAM11,          0xa4 },
+	{ OV7725_GAM12,          0xaf },
+	{ OV7725_GAM13,          0xc5 },
+	{ OV7725_GAM14,          0xd7 },
+	{ OV7725_GAM15,          0xe8 },
+	{ OV7725_SLOP,           0x20 },
+
+	{ OV7725_COM3,           0x40 },
+	{ OV7725_COM5,           0xf5 },
+	{ OV7725_COM10,          0x02 },
+	{ OV7725_COM2,           0x01 }
+};
+
+static int ov7725_write_reg(const struct device *dev, uint8_t reg_addr,
+			    uint8_t value)
+{
+	struct ov7725_data *drv_data = dev->data;
+	struct i2c_msg msgs[2];
+
+	msgs[0].buf = (uint8_t *)&reg_addr;
+	msgs[0].len = 1;
+	msgs[0].flags = I2C_MSG_WRITE;
+
+	msgs[1].buf = (uint8_t *)&value;
+	msgs[1].len = 1;
+	msgs[1].flags = I2C_MSG_WRITE | I2C_MSG_STOP;
+
+	return i2c_transfer(drv_data->i2c, msgs, 2, drv_data->i2c_addr);
```

Comment:
Is there a reason you can't use `i2c_reg_write_byte()` instead?

Reply:
Although i2c_reg_write_byte can achieve the same purpose, I hope it is the similar as ov7725_read_reg.